src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,18 +29,23 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={filterInputValue}
+        onChange={event => setFilterInputValue(event.currentTarget.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
       <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
+        {filterInputValue && (
```

Comment:
```suggestion
        {!!filterInputValue && (
```
in rendering with && for best rendering control use !! to return a boolean value